tests/metrics/test_heuristics_metrics.py

```diff
@@ -91,6 +101,43 @@ def test_randomization_metric(
     out = metric.compute()["score"]
     assert (out >= -1.0) & (out <= 1.0), "Test failed."
 
+    # 2) Check if the randomization works correctly
+    batch_size = 2
+    input_shape = test_data[0].shape
+    random_tensor = torch.randn((batch_size, *input_shape), device="cpu")
+
+    rand_model = metric._randomize_model()[0]
+    rand_model.eval()
+    model.eval()
+
+    with torch.no_grad():
+        original_out = model(random_tensor)
+        randomized_out = rand_model(random_tensor)
+
+    assert not torch.allclose(
+        original_out, randomized_out
+    ), "Outputs do not differ after randomization"
+    assert not torch.isnan(
+        randomized_out
+    ).any(), "Randomized model output contains NaNs."
```

Comment:
Could we make 2) and 3) separate tests? In general, we aim to have only 1 assert per test. This makes debugging easier.